src/activity_handlers.cpp

```diff
@@ -2632,8 +2630,24 @@ void activity_handlers::try_sleep_finish( player_activity *act, player *p )
 
 void activity_handlers::craft_do_turn( player_activity *act, player *p )
 {
-    const recipe &rec = recipe_id( act->name ).obj();
+    item *craft = act->targets.front().get_item();
+
+    if( !craft->is_craft() ) {
+        debugmsg( "ACT_CRAFT target '%s' is not a craft.  Aborting ACT_CRAFT.", craft->tname() );
+        p->cancel_activity();
+        return;
+    }
+    if( !p->has_item( *craft ) ) {
+        p->add_msg_if_player( "%s no longer has the target '%s.'  Aborting ACT_CRAFT.",
+                              p->disp_name(), craft->tname() );
```

Comment:
This can appear during normal play, so it shouldn't look like a debug message.

Reply:
Pretty sure I intended for that to be a debug message but called the wrong function.  I guess it could pop up rarely in normal play however and wouldn't cause any problems, so I'll take your advice and fix it.

Thanks for all your time reviewing this pr by the way!

Reply:
Thanks in return for your efforts working on it: the list of presumably resolved issues is quite impressive.